### ISSUE TITLE
Warning 41 is misleading

### ISSUE THREAD

**Issue by Chris00** [AltGr/opam-ci#18]
[Warning 41](https://github.com/ocaml/opam-repository/pull/10683)

> warning 41: Some packages are mentionned in package scripts of features,
but there is no dependency or depopt toward them: "user-setup"

seem to imply that _user-setup_ should be mentioned as a dependency — and indeed doing so makes the warning go away — but doing so creates a circular dependency.  An improvement would be welcome.

**Comment by AltGr** [comment]
Indeed, it's often likely an error, but not in this case. Probably `post-messages`, which is not evaluated in the scope of the package but in the final state of the install, shouldn't be considered for warning 41.

**Comment by AltGr** [comment]
This has actually been fixed already in opam, which I forgot; not backported yet to Camelus / 1.2.2 format though. Thanks for reporting!
